SmartDeviceLinkTests/DevAPISpecs/SDLPreloadChoicesOperationSpec.m

```diff
@@ -297,6 +308,121 @@
+                    SDLTextField *primaryTextField = [[SDLTextField alloc] init];
+                    primaryTextField.name = SDLTextFieldNameMenuName;
+                    windowCapability.textFields = @[primaryTextField];
+
+                    testOp = [[SDLPreloadChoicesOperation alloc] initWithConnectionManager:testConnectionManager fileManager:testFileManager displayName:testDisplayName windowCapability:windowCapability isVROptional:NO cellsToPreload:testCells updateCompletionHandler:^(NSArray<NSNumber *> * _Nullable failedChoiceUploadIDs) {}];
+                    [testOp start];
+                    [testOp removeChoicesFromUpload:[NSSet setWithArray:@[testCell1]]];
+
+                    NSArray<SDLCreateInteractionChoiceSet *> *receivedRequests = (NSArray<SDLCreateInteractionChoiceSet *> *)testConnectionManager.receivedRequests;
+
+                    expect(receivedRequests).to(haveCount(2));
+                    expect(receivedRequests[0].choiceSet[0].menuName).to(equal(testCell1.text));
+                    expect(receivedRequests[1].choiceSet[0].menuName).to(equal(testCell2.text));
+                });
+            });
+        });
+
+        context(@"the module's response to choice uploads", ^{
+            __block SDLChoiceCell *testCell1 = nil;
+            __block SDLChoiceCell *testCell2 = nil;
+            __block NSOrderedSet<SDLChoiceCell *> *testCells = nil;
+            __block SDLCreateInteractionChoiceSetResponse *testBadResponse = nil;
+            __block SDLCreateInteractionChoiceSetResponse *testGoodResponse = nil;
+
+            beforeEach(^{
+                testCell1 = [[SDLChoiceCell alloc] initWithText:@"Cell1" artwork:nil voiceCommands:nil];
+                testCell1.choiceId = 55;
+                testCell2 = [[SDLChoiceCell alloc] initWithText:@"Cell2" secondaryText:nil tertiaryText:nil voiceCommands:nil artwork:nil secondaryArtwork:[SDLArtwork artworkWithStaticIcon:SDLStaticIconNameClock]];
+                testCell2.choiceId = 66;
+                testCells = [[NSOrderedSet alloc] initWithArray:@[testCell1, testCell2]];
+
+                testBadResponse = [[SDLCreateInteractionChoiceSetResponse alloc] init];
+                testBadResponse.success = @NO;
+                testBadResponse.resultCode = SDLResultRejected;
+
+                testGoodResponse = [[SDLCreateInteractionChoiceSetResponse alloc] init];
+                testGoodResponse.success = @YES;
+                testGoodResponse.resultCode = SDLResultSuccess;
+            });
+
+            context(@"when a bad response comes back", ^{
+                 it(@"should add the choiceID of the failed choice item to the failedChoiceUploadIDs array", ^{
+                    SDLTextField *primaryTextField = [[SDLTextField alloc] init];
+                    primaryTextField.name = SDLTextFieldNameMenuName;
+                    windowCapability.textFields = @[primaryTextField];
```

Comment:
This seems to be the same setup as above (below as well) so maybe the setup should go even higher up than my previous comment described?

Reply:
I removed the code as it was not necessary for the test to run.